File: sem3/map/Lab/toylanguage_C#/ToyLanguage/ToyLanguage/Models/Expressions/ConstantExpression.cs
using ToyLanguage.Models.Interfaces;
using ToyLanguage.Models.States;

namespace ToyLanguage.Models.Expressions
{
    public class ConstantExpression : IExpression
    {
        private readonly int _value;

        public ConstantExpression(int value)
        {
            _value = value;
        }

        public int Eval(ProgramState state)
        {
            return _value;
        }

        public override string ToString()
        {
            return _value.ToString();
        }
    }
}

File: sem3/map/Exam/GUI_ToyLanguageInterpreter/src/Interpreter/Models/Expressions/ConstantExpression.java
package Interpreter.Models.Expressions;

import Interpreter.Models.Expression;
import Interpreter.Models.ProgramState.ProgramState;

public class ConstantExpression extends Expression {
    private int value;

    public ConstantExpression(int value) {
        this.value = value;
    }

    @Override
    public int eval(ProgramState state) {
        return this.value;
    }

    @Override
    public String toString() {
        return ""+this.value;
    }
}
